src/AppInstallerCommonCore/AppInstallerTelemetry.cpp

```diff
@@ -87,13 +87,28 @@ namespace AppInstaller::Logging
         m_isRuntimeEnabled = true;
     }
 
+    void TelemetryTraceLogger::SetUserSettingsStatus()
+    {
+        m_isSettingEnabled = !Settings::User().Get<Settings::Setting::TelemetryDisable>();
+    }
+
+    void TelemetryTraceLogger::SetTelemetryCorelationJson(std::wstring jsonStr)
+    {
+        m_telemetryCorelationJson = jsonStr;
+    }
+
+    void TelemetryTraceLogger::SetCOMCaller(std::wstring comCaller)
+    {
+        m_comCaller = comCaller;
+    }
+
     void TelemetryTraceLogger::LogFailure(const wil::FailureInfo& failure) const noexcept
     {
         if (IsTelemetryEnabled())
         {
-            TraceLoggingWriteActivity(g_hTelemetryProvider,
+            TraceLoggingWriteActivity(g_hTraceProvider,
```

Comment:
Why update all of these lines just to change the global name?  What value is there in changing the name of the global?

Reply:
The value is not in any better functionality but in it communicating truly what it is. The provider is NOW not only used for logging Telemetry but Diagnostic events as well.